apps/studio/src/components/PageEditor/ComponentSelector.tsx

```diff
@@ -95,12 +98,33 @@ function BlockItem({
 }
 
 function ComponentSelector() {
-  const { setDrawerState, setPageState, setEditorState } =
+  const { setCurrActiveIdx, pageState, setDrawerState, setPageState } =
     useEditorDrawerContext()
+  const { mutate } = trpc.page.updatePageBlob.useMutation()
```

Comment:
should you handle the loading state too?

Reply:
sure, will add!

Reply:
actually going in the opposite direction here - not doing a save at all. the reason why is because we don't want empty states to show up in the end user site - ie, i should save only when the user has content **that they added**. confirmed with @sehyunidaaa already so if they add a block (no content) then refresh, we're ok with this disappearing